demo-invenioils/docker/frontend/Dockerfile

```
@@ -1,21 +1,28 @@
-FROM node:14 AS react-build
+ARG BACKEND_BUILT_IMAGE_URL
 
-ENV WORKING_DIR=/code/ui
-RUN mkdir -p ${WORKING_DIR}
-COPY ./ui ${WORKING_DIR}
-WORKDIR ${WORKING_DIR}
+# Import invenio built image
+FROM $BACKEND_BUILT_IMAGE_URL as invenio-build
 
-# build React application
-RUN npm install
-RUN REACT_APP_INVENIO_UI_URL="https://127.0.0.1" \
-    REACT_APP_INVENIO_REST_ENDPOINTS_BASE_URL="https://127.0.0.1/api" \
-    REACT_APP_ENV_NAME="dev" \
-    npm run build
+FROM node:14 as react-build
 
-FROM nginx:1.18
+ENV WORKING_DIR=/code
+COPY ./ ${WORKING_DIR}
 
-COPY docker/frontend/nginx.conf /etc/nginx/nginx.conf
-COPY docker/frontend/conf.d/* /etc/nginx/conf.d/
+ARG ENV_NAME
+ARG INVENIO_UI_URL
+ARG INVENIO_REST_ENDPOINTS_BASE_URL
+ENV REACT_APP_INVENIO_UI_URL=$INVENIO_UI_URL
+ENV REACT_APP_INVENIO_REST_ENDPOINTS_BASE_URL=$INVENIO_REST_ENDPOINTS_BASE_URL
+ENV REACT_APP_ENV_NAME=$ENV_NAME
+
+# build ui react application
+RUN cd ui && npm install && \
+ REACT_APP_INVENIO_UI_URL=$INVENIO_UI_URL \
+ REACT_APP_INVENIO_REST_ENDPOINTS_BASE_URL=$INVENIO_REST_ENDPOINTS_BASE_URL \
+ REACT_APP_ENV_NAME=$ENV_NAME \
+ npm run build
+
+FROM nginx:1.21
```

```suggestion
FROM registry.cern.ch/docker.io/library/nginx:1.21
```
To avoid hitting docker rate limit (limited to 100 pulls per 6 hours if I am not wrong)